mev-relay-rs/src/relay.rs

```diff
@@ -448,6 +451,28 @@ impl Relay {
         state.auctions.insert(auction_request, auction_context);
         Ok(())
     }
+
+    fn verify_blinded_block_signature(
+        &self,
+        signed_block: &mut SignedBlindedBeaconBlock,
+    ) -> Result<(), ethereum_consensus::Error> {
+        let proposer = self
+            .validator_registry
+            .get_public_key(signed_block.message().proposer_index())
+            .unwrap();
```

Comment:
2 things:

1) we already have the proposer's public key in the `AuctionRequest`, consider just passing this is as a parameter to this function
2) you'll won't need to worry about this after (1) but we would not `unwrap` here as if the key is missing it would just panic :(    instead, you'd want to return some kind of error